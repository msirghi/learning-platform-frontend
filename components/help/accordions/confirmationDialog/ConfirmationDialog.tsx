import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../i18n';

type Props = {
  open: boolean;
  handleClose: () => void;
  searchInputValue: string;
  onDialogSubmit: () => void;
};

export const QuestionConfirmationDialog: React.FC<Props> = ({
  open,
  handleClose,
  searchInputValue,
  onDialogSubmit
}) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        maxWidth='lg'
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('help:confirmationDialog.title')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            <div>
              <div>{t('help:confirmationDialog.messageWillBeSentTitle')}:</div>
              <div>
                <b>{searchInputValue}</b>
              </div>
              <div>{t('help:confirmationDialog.confirmYourAction')}</div>
            </div>
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button variant='outlined' onClick={handleClose} color='secondary'>
            {t('help:confirmationDialog.cancelButtonLabel')}
          </Button>
          <Button variant='outlined' onClick={onDialogSubmit} color='primary'>
            {t('help:confirmationDialog.confirmButtonLabel')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
