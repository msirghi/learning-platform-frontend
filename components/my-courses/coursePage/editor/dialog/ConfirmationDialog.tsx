import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../../i18n';

type Props = {
  message: string;
  onSubmit: () => void;
  onCancel: () => void;
  open: boolean;
};

export const ConfirmationDialog: React.FC<Props> = ({ open, message, onSubmit, onCancel }) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        open={open}
        onClose={onCancel}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('courses:confirmationDialogTitle')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>{message}</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onCancel} color='secondary' variant='outlined'>
            {t('courses:cancel')}
          </Button>
          <Button onClick={onSubmit} color='primary' variant='outlined'>
            {t('courses:continue')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
