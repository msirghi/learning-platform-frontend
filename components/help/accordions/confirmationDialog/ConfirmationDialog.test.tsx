import { Button, Dialog } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { QuestionConfirmationDialog } from './ConfirmationDialog';

describe('QuestionConfirmationDialog component', () => {
  const defaultProps = {
    open: false,
    handleClose: jest.fn(),
    searchInputValue: 'input value',
    onDialogSubmit: jest.fn()
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<QuestionConfirmationDialog {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleClose prop on cancel button click', () => {
    const wrapper = shallow(<QuestionConfirmationDialog {...defaultProps} />);
    const button = wrapper.find(Button).at(0);

    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(defaultProps.handleClose).toBeCalled();
  });

  it('should call handleClose prop on dialog close', () => {
    const wrapper = shallow(<QuestionConfirmationDialog {...defaultProps} />);
    const dialog = wrapper.find(Dialog);
    expect(dialog).toHaveLength(1);

    dialog.props().onClose({}, 'backdropClick');
    expect(defaultProps.handleClose).toBeCalled();
  });

  it('should call onDialogSubmit prop on submit button click', () => {
    const wrapper = shallow(<QuestionConfirmationDialog {...defaultProps} />);
    const button = wrapper.find(Button).at(1);

    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(defaultProps.onDialogSubmit).toBeCalled();
  });

  it('should render provided message', () => {
    const wrapper = shallow(<QuestionConfirmationDialog {...defaultProps} />);
    expect(wrapper.text().includes(defaultProps.searchInputValue)).toBeTruthy();
  });
});
