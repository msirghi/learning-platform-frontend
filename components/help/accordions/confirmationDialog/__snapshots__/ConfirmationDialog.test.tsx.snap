// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionConfirmationDialog component should match the snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    maxWidth="lg"
    onClose={[MockFunction]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="alert-dialog-title"
    >
      help:confirmationDialog.title
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogContentText))
        id="alert-dialog-description"
      >
        <div>
          <div>
            help:confirmationDialog.messageWillBeSentTitle
            :
          </div>
          <div>
            <b>
              input value
            </b>
          </div>
          <div>
            help:confirmationDialog.confirmYourAction
          </div>
        </div>
      </WithStyles(ForwardRef(DialogContentText))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        onClick={[MockFunction]}
        variant="outlined"
      >
        help:confirmationDialog.cancelButtonLabel
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[MockFunction]}
        variant="outlined"
      >
        help:confirmationDialog.confirmButtonLabel
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
